src/_guides/openapi/specification/v3.1/the-perfect-modern-openapi-workflow.md

```diff
@@ -0,0 +1,660 @@
+> Laravel PHP](_guides/openapi/design-first-laravel-php.md). If you'd like to
+> see a similar guide for your favorite language/framework please [get in
+> touch](mailto:hello@bump.sh).
+{: .info }
+
+If an API does not have an existing integration test suite this might seem like
+a bigger push, but an API without a test suite should absolutely add one.
+Working with a generic test suite and adding in some OpenAPI assertions can be a
+great way to start off a larger test suite. Set up one HTTP request for each API
+endpoint, with basic information, and add other tests for various scenarios over
+time as bugs are squashed.
+
+Another option to avoid that is to run contract testing outside of the codebase.
+To avoid having to train a tool to know what the expected contract is, why not
+use a tool which already knows what the latest OpenAPI is meant to be at
+anytime: Microcks again! 
+
+Microcks handles [contract
+testing](_guides/bump-sh-tutorials/testing-with-microcks.md) as well as mock
+servers, and it does this by taking a URL to a server for comparison. This could
+be staging, pre-production, or even production if you're careful. 
+
+It works by going through all the operations in the OpenAPI document, and uses
+the examples and schemas defined there to send a request that should work, to an
+API instance of your choosing. This could be production if you are brave, or
+some other staging/testing environment, but the logic is simple:
+
+1. Send HTTP requests.
+2. See if that fails unexpectedly. 
+3. Receive HTTP response.
+4. See if that matches the OpenAPI document.
+
+This could be automated to run at regular intervals, or it could be triggered to
+run on pull requests and merges to make sure that any and all API or OpenAPI
+changes agree with each other. GitHub Actions is once again a good way to get
+this done.
+
+```yaml
+# .github/workflows/contract-testing.yml
+name: API Contract Testing
+on: [push]
+jobs:
+  contract-testing:
+    name: Deploy API documentation on Bump.sh
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - uses: microcks/import-github-action@v1
+        with:
+          specificationFiles: 'api/openapi.yaml:true'
+          microcksURL: 'https://mocks.example.com/api/'
+          keycloakClientId:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT }}
+          keycloakClientSecret:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT_CREDENTIALS }}
+
+      - uses: microcks/test-github-action@v1
+        with:
+          apiNameAndVersion: 'Train Travel API:1.0.0'
+          testEndpoint: 'http://api-testing.example.com'
+          runner: OPEN_API_SCHEMA
+          microcksURL: 'https://mocks.example.com/api/'
+          keycloakClientId:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT }}
+          keycloakClientSecret:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT_CREDENTIALS }}
+          waitFor: '10sec'
+```
+
+This can replace the `.github/workflows/api-mocks.yaml` workflow above which
```

Comment:
I don't see a `api-mocks.yaml` workflow “above” was this forgotten?

Reply:
ah! [found it below](https://github.com/bump-sh/docs/pull/330/files#diff-ef5d127a86aa8c49a4dfd89c764bb9e1c2a85be973803893728604d5e5ec22b9R507) and it's called `api-mocks.yml` :grin: